src/docs/markdown/caddyfile/options.md

```diff
@@ -83,27 +123,84 @@ Specifies an External Account Binding to use for all ACME transactions.
+		protocol {
+			allow_h2c
+		}
+	}
+}
+```
+
+##### `listener_wrappers`
+
+Allows configuring [listener wrappers](/docs/json/apps/http/servers/listener_wrappers/), which can modify the behaviour of the base listener. They are applied in the given order.
+
+There is a special no-op [`tls`](/docs/json/apps/http/servers/listener_wrappers/tls/) listener wrapper provided as a standard module which marks where TLS should be handled in the chain of listener wrappers. It should only be used if another listener wrapper must be placed in front of the TLS handshake.
+
+For example, assuming you have the [`proxy_protocol`](/docs/json/apps/http/servers/listener_wrappers/proxy_protocol/) plugin installed:
+
+```caddy-d
+listener_wrappers {
+	proxy_protocol {
+		timeout 2s
+		allow 192.168.86.1/24 192.168.86.1/24
+	}
+	tls
+}
+```
+
+##### `timeouts`
+
+- **read_timeout** is a [duration value](/docs/conventions#durations) that sets how long to allow a read from a client's upload. Setting this to a short, non-zero value can mitigate slowloris attacks, but may also affect legitimately slow clients. Defaults to no timeout.
+
+- **read_header_timeout** is a [duration value](/docs/conventions#durations) that sets how long to allow a read from a client's request headers. Defaults to no timeout.
+
+- **write_timeout** is a [duration value](/docs/conventions#durations) that sets how long to allow a write to a client. Note that setting this to a small value when serving large files may negatively affect legitimately slow clients. Defaults to no timeout.
+
+- **idle_timeout** is a [duration value](/docs/conventions#durations) that sets the maximum time to wait for the next request when keep-alives are enabled. Defaults to 5 minutes to help avoid resource exhaustion.
+
+##### `max_header_size`
+The maximum size to parse from a client's HTTP request headers. It accepts all formats supported by [go-humanize](https://github.com/dustin/go-humanize/blob/master/bytes.go).
+
+##### `protocol`
+
+- **allow_h2c** enables H2C ("Cleartext HTTP/2" or "H2 over TCP") support, which will serve HTTP/2 over plaintext TCP connections if a client support it. Because this is not implemented by the Go standard library, using H2C is incompatible with most of the other options for this server. Do not enable this only to achieve maximum client compatibility. In practice, very few clients implement H2C, and even fewer require it. This setting applies only to unencrypted HTTP listeners. ⚠️ Experimental feature; subject to change or removal.
+
+- **experimental_http3** enables experimental draft HTTP/3 support. Note that HTTP/3 is not a finished spec and client support is extremely limited. This option will go away in the future. _This option is not subject to compatibility promises._
+
+- **strict_sni_host** require that a request's `Host` header matches the value of the ServerName sent by the client's TLS ClientHello; often a necessary safeguard when using TLS client authentication.
```

Comment:
Come to think of it, it's a little weird that this is under protocol, because this doesn't actually configure protocols.

Reply:
I had this argument with @whitestrake months ago when we were hashing out the original plan, and the conclusion we got to was "this is a TLS thing, which is protocol" but yeah it's fuzzy.

Reply:
I think this is more of a TLS option, since it does require TLS to make sense.

Reply:
Sure, but it's still a per-server option though, in JSON. I don't think it's worth making a new `tls` section for this in `servers`, nor do I think it should be an orphan. :man_shrugging: